app/models/metric/ci_mixin/capture.rb

```diff
@@ -32,27 +46,22 @@ def perf_capture_queue(interval_name, options = {})
 
     log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
 
-    # Determine what items we should be queuing up
-    items = []
     cb = nil
     if interval_name == 'historical'
       start_time = Metric::Capture.historical_start_time if start_time.nil?
       end_time = Time.now.utc if end_time.nil?
-
-      start_hour = start_time
-      while start_hour != end_time
-        end_hour = start_hour + 1.day
-        end_hour = end_time if end_hour > end_time
-        items.unshift([interval_name, start_hour, end_hour])
-        start_hour = end_hour
-      end
     else
-      items << [interval_name]
-      items[0] << start_time << end_time unless start_time.nil?
-
+      start_time = last_perf_capture_on unless start_time
```

Comment:
> > This change only helps if the caller uses the queue to queue up the perf capture.
>Is there any place that doesn't do this?

I *believe* all of the important parts of the code queue today, but there's still some method name building that's hard to track down.

> So, are you saying that the provider-specific part of perf_capture should be responsible for determining what to do with the timestamps given to them? If so, I thought that's what the code already does (aside from the gap collection).

No, I'm saying that there are [provider-specific implementations](https://github.com/ManageIQ/manageiq/blob/c2f76e9eec052c5a8aa0cb8d2b56f9ce98eba48d/app/models/manageiq/providers/openstack/base_metrics_capture.rb#L5-L8) that still expect that the caller can send in `nil` for start/end dates.  And, I think the interface should be more strict to assume that the caller *always* provides a valid start/end date.  Instead of allowing the assumption of what a `nil` start/end date means permeate throughout the metrics capture code.  In the end, if no one is calling specific implementations with `nil` start/end date, then it's wasted logic.  And, if people *are* calling specific implementations with `nil` start/end date, then it's a pattern of deviation that will show up as we consolidate more things into the provider platform arena.